omnigibson/prims/xform_prim.py

```diff
@@ -169,47 +170,114 @@ def has_material(self):
+                if "xformOp:orient" not in properties:
+                    lazy.carb.log_error(
+                        "Orient property needs to be set for {} before setting its orientation".format(self.name)
+                    )
+                xform_op = self._prim.GetAttribute("xformOp:orient")
+                if xform_op.GetTypeName() == "quatf":
+                    rotq = lazy.pxr.Gf.Quatf(*orientation)
+                else:
+                    rotq = lazy.pxr.Gf.Quatd(*orientation)
+                xform_op.Set(rotq)
+            PoseAPI.invalidate()
+            if gm.ENABLE_FLATCACHE:
+                # If flatcache is on, make sure the USD local pose is synced to the fabric local pose.
+                # Ideally we should call usdrt's set local pose directly, but there is no such API.
+                # The only available API is SetLocalXformFromUsd, so we update USD first, and then sync to fabric.
+                xformable_prim = lazy.usdrt.Rt.Xformable(
+                    lazy.omni.isaac.core.utils.prims.get_prim_at_path(self.prim_path, fabric=True)
+                )
+                assert (
+                    not xformable_prim.HasWorldXform()
+                ), "Fabric's world pose is set for a non-rigid prim which is unexpected. Please report this."
+                xformable_prim.SetLocalXformFromUsd()
 
-    def get_position_orientation(self):
+    def get_position_orientation(self, frame: Literal["world", "scene", "parent"] = "world"):
         """
-        Gets prim's pose with respect to the world's frame.
+        Gets prim's pose with respect to the specified frame.
+
+        Args:
+            frame (Literal): frame to get the pose with respect to. Default to world. parent frame
+            get position relative to the object parent. scene frame get position relative to the scene.
 
         Returns:
             2-tuple:
-                - 3-array: (x,y,z) position in the world frame
-                - 4-array: (x,y,z,w) quaternion orientation in the world frame
+                - 3-array: (x,y,z) position in the specified frame
+                - 4-array: (x,y,z,w) quaternion orientation in the specified frame
         """
-        return PoseAPI.get_world_pose(self.prim_path)
+
+        if frame == "parent":
+            return PoseAPI.get_position_orientation(self.prim_path, frame)
+        else:
+            position, orientation = PoseAPI.get_position_orientation(self.prim_path, "world")
+
+            # If we are in a scene, compute the scene-local transform (and save this as the world transform
+            # for legacy compatibility)
+            if frame == "scene" and self.scene is not None:
+                position, orientation = T.relative_pose_transform(
+                    position, orientation, *self.scene.prim.get_position_orientation()
+                )
+
+            return position, orientation
+
+    # ------------------- Deprecated methods -------------------
```

Comment:
nit: is this the correct way to specify "deprecated methods"? Is there a formal / better way?